elastic/kibana
# [ILM] Fix data allocation field

Fixes https://github.com/elastic/kibana/issues/173764

## Summary

This PR fixes the "Custom" data allocation field in the edit IL policy form by adding a default value for the field. Before, there was no default value, so no value would be added to the request until the user changes the initial selection to another option. This explains the bug described from  https://github.com/elastic/kibana/issues/173764 (in v7.17) where there is only one option and the user is not able to change it to something else so the value of the only option cannot be added to the request. Now with these changes, if the user doesn't make any changes to the initially selected option, the default value would be added to the request.

**How to test:**
1. Start Es with `yarn es snapshot` and Kibana with `yarn start`.
2. Go to Stack Management -> Index Lifecycle Policies and start creating a new policy.
3. Type in some name for the policy. Enable the Warm (or Cold) phase and type in some number in the "Move data into phase when"  field. Click on "Advanced settings".
4. In the "Data allocation" field, select the "Custom" option.
5. Click on "Show request" and verify that the request correctly shows the `allocation` field (the "Any data node" option doesn't add anything to the request).
6. Select another option for the node attribution field and verify it is displayed in the request.

Test the field on cloud by accessing the [CI cloud deployment](https://kibana-pr-176292.kb.us-west2.gcp.elastic-cloud.com:9243/) from this PR or by changing the if-condition in line 65 in `x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/phases/shared_fields/data_tier_allocation_field/components/node_allocation.tsx` to `if (false)`.  Now the "Any data node" option is not available. Follow the test instructions above and verify that the initially selected node attribute option is added to the request.
